### ISSUE TITLE
`NonZero<Unsigned>::get() - 1` should not trigger `arithmetic-side-effects`

### ISSUE THREAD

**Issue by demosdemon** [rust-lang/rust-clippy#15225]
### Summary

`NonZero<_>::get()` is guaranteed to produce a non-zero value. Therefore, subtracting one from that value is trivially not going to panic or wrap.

### Lint Name

arithmetic-side-effects

### Reproducer

I tried this code:

```rust
#[deny(clippy::arithmetic_side_effects)]
pub fn do_something(v: NonZeroUsize) {
    let _n = v.get() - 1;
    todo!()
}
```

I saw this happen:

```
error: arithmetic operation that can potentially result in unexpected side-effects
 --> src/lib.rs:5:14
  |
5 |     let _n = v.get() - 1;
  |              ^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#arithmetic_side_effects
```

I expected to see this happen:

No output as `NonZero<_>::get() - 1` is trivially safe.

### Version

```text
1.90.0-nightly (a84ab0ce6 2025-07-06)
binary: rustc
commit-hash: a84ab0ce6c4557a2f01a3a6c3fdb0f92098db78d
commit-date: 2025-07-06
host: aarch64-apple-darwin
release: 1.90.0-nightly
LLVM version: 20.1.7
```
